server: fix over redacted log line in migration server

aa-joshi opened the pull request:
Previosuly, few log lines were getting redacted assuming that it contains
sensitive information. Support team was facing challenges during dignostics. To
address this, this patch address the log line in migration server which was
overly redacted.

Epic: CRDB-37533
Part of: CRDB-44885
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/137915)
<!-- Reviewable:end -->

arjunmahishi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137915)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi)

___
*[`pkg/server/migration.go` line 147 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/137915#-OEn7zBj4sxHBYcGfhng:-OEn7zBj4sxHBYcGfhnh:b22bhaz) ([raw file](https://github.com/cockroachdb/cockroach/blob/430dd7902a79a21efb35c2d13c8c8c014c8a2948/pkg/server/migration.go#L147)):*
> ```Go
> 	}
> 	log.Infof(ctx, "active cluster version setting is now %s (up from %s)",
> 		redact.Safe(newCV.PrettyPrint()), redact.Safe(prevCV.PrettyPrint()))
> ```

I think we can directly pass `newCV` and `prevCV` without `Safe` or `PrettyPrint`. Because the underlying type implements `SafeFormatter`

It was getting redacted because `PrettyPrint()` returns `string`


<!-- Sent from Reviewable.io -->

aa-joshi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137915)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi)

___
*[`pkg/server/migration.go` line 147 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/137915#-OEn7zBj4sxHBYcGfhng:-OEn9l-S0BXUBlW_q9XB:bsr6zyp) ([raw file](https://github.com/cockroachdb/cockroach/blob/430dd7902a79a21efb35c2d13c8c8c014c8a2948/pkg/server/migration.go#L147)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

I think we can directly pass `newCV` and `prevCV` without `Safe` or `PrettyPrint`. Because the underlying type implements `SafeFormatter`

It was getting redacted because `PrettyPrint()` returns `string`
</blockquote></details>

I believe we need to wrap in PrettyPrint because it has formatting logic based on `v.IsFence()`
ref: https://github.com/cockroachdb/cockroach/blob/ac3c5f2e66f0bb3c992c719389f15d855248a781/pkg/roachpb/version.go#L115


<!-- Sent from Reviewable.io -->

arjunmahishi commented:
*[`pkg/server/migration.go` line 147 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/137915#-OEn7zBj4sxHBYcGfhng:-OEnApWpBOORNs8QMYhl:b-5lj7ri) ([raw file](https://github.com/cockroachdb/cockroach/blob/430dd7902a79a21efb35c2d13c8c8c014c8a2948/pkg/server/migration.go#L147)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

I believe we need to wrap in PrettyPrint because it has formatting logic based on `v.IsFence()`
ref: https://github.com/cockroachdb/cockroach/blob/ac3c5f2e66f0bb3c992c719389f15d855248a781/pkg/roachpb/version.go#L115
</blockquote></details>

Then I think we can change PrettyPrint and make it return `redact.RedactableString`
I think it's only being used for logging anyway. 


<!-- Sent from Reviewable.io -->

aa-joshi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137915)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi)

___
*[`pkg/server/migration.go` line 147 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/137915#-OEn7zBj4sxHBYcGfhng:-OFaWofE7THpklYw1puD:b-ls9wvq) ([raw file](https://github.com/cockroachdb/cockroach/blob/430dd7902a79a21efb35c2d13c8c8c014c8a2948/pkg/server/migration.go#L147)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

Then I think we can change PrettyPrint and make it return `redact.RedactableString`
I think it's only being used for logging anyway.
</blockquote></details>

PrettyPrint is invoked at multiple flows. Changing return type would increase changes significantly. As part of the task, we want to keep changes minimal so that we can backport. Let's use `SafeString` here.
___
*[`pkg/server/auto_upgrade.go` line 52 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/137915#-OFaXJTu86NbQQigQvvY:-OFaXJTu86NbQQigQvvZ:b-v005jk) ([raw file](https://github.com/cockroachdb/cockroach/blob/430dd7902a79a21efb35c2d13c8c8c014c8a2948/pkg/server/auto_upgrade.go#L52)):*
> ```Go
> 			clusterVersion, err := s.clusterVersion(ctx)
> 			if err != nil {
> 				log.Errorf(ctx, "unable to retrieve cluster version: %v", redact.Safe(err))
> ```

I have reverted log redaction changes for errors as we have decided to build a standard framework for it.


<!-- Sent from Reviewable.io -->

rafiss reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137915)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi)

___
*[`pkg/server/migration.go` line 147 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/137915#-OEn7zBj4sxHBYcGfhng:-OFxmRz3CG2P5gaZZkwP:b-4nm6jv) ([raw file](https://github.com/cockroachdb/cockroach/blob/430dd7902a79a21efb35c2d13c8c8c014c8a2948/pkg/server/migration.go#L147)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

PrettyPrint is invoked at multiple flows. Changing return type would increase changes significantly. As part of the task, we want to keep changes minimal so that we can backport. Let's use `SafeString` here.
</blockquote></details>

That change does not significantly change any of the non-test call sites. I wrote up a quick PR here to show it: https://github.com/cockroachdb/cockroach/pull/138353

IMO we should change PrettyPrint to return a RedactableString rather than using SafeString here.


<!-- Sent from Reviewable.io -->

rafiss reviewed (changes requested):
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137915)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi)



<!-- Sent from Reviewable.io -->

aa-joshi committed:
server: fix over redacted log line in migration server

Previosuly, few log lines were getting redacted assuming that it contains
sensitive information. Support team was facing challenges during dignostics. To
address this, this patch address the log line in migration server which was
overly redacted.

Epic: CRDB-37533
Part of: CRDB-44885
Release note: None

aa-joshi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137915)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi and @rafiss)

___
*[`pkg/server/migration.go` line 147 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/137915#-OEn7zBj4sxHBYcGfhng:-OFzUPTOBKD1OBdIRt8u:b-d971ha) ([raw file](https://github.com/cockroachdb/cockroach/blob/430dd7902a79a21efb35c2d13c8c8c014c8a2948/pkg/server/migration.go#L147)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

That change does not significantly change any of the non-test call sites. I wrote up a quick PR here to show it: https://github.com/cockroachdb/cockroach/pull/138353

IMO we should change PrettyPrint to return a RedactableString rather than using SafeString here.
</blockquote></details>

Thanks for the review! updated.


<!-- Sent from Reviewable.io -->

arjunmahishi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137915)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi and @rafiss)

___
*[`pkg/clusterversion/cockroach_versions_test.go` line 150 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/137915#-OG-E-B6A5Mq4Dxt9E-w:-OG-E-B6A5Mq4Dxt9E-x:bo4h2u9) ([raw file](https://github.com/cockroachdb/cockroach/blob/4570e95d3faf16bf2020dd2eb3ab15223166318d/pkg/clusterversion/cockroach_versions_test.go#L150)):*
> ```Go
> 	}
> 	for _, test := range tests {
> 		if actual := test.cv.PrettyPrint().StripMarkers(); actual != test.exp {
> ```

We should not be getting any redaction markers if the redaction is handled correctly. If we are just using `StripMarkers` to resolve the type to string, we can just convert the type of `exp` to `redact.RedactableString` on line 141.

Removing `StripMarkers` will ensure that the test asserts the output of `PrettyPrint` is not redacted. 


<!-- Sent from Reviewable.io -->

rafiss reviewed (approved):
nice fix!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137915#-:-OG0RMtuCdBcfJfI3D0Q:bhpddey)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi)

___
*[`pkg/clusterversion/cockroach_versions_test.go` line 150 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/137915#-OG-E-B6A5Mq4Dxt9E-w:-OG0QuPv4P3HKEPZRcgX:bllbig4) ([raw file](https://github.com/cockroachdb/cockroach/blob/4570e95d3faf16bf2020dd2eb3ab15223166318d/pkg/clusterversion/cockroach_versions_test.go#L150)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

We should not be getting any redaction markers if the redaction is handled correctly. If we are just using `StripMarkers` to resolve the type to string, we can just convert the type of `exp` to `redact.RedactableString` on line 141.

Removing `StripMarkers` will ensure that the test asserts the output of `PrettyPrint` is not redacted.
</blockquote></details>

in tests, there's no strong reason to use redacted strings. IMO, it's ok to use StripMarkers here. however, it's also true that changing the type of `exp` should fix it and is also a simple change. i'd be fine either way,


<!-- Sent from Reviewable.io -->

aa-joshi commented:
TFTR
bors r+

aa-joshi commented:
I am skipping backporting to 23.2 considering implementation changes.
